### ISSUE TITLE
Runs fine on Replit but not on M1 mac

### ISSUE THREAD

**Issue by chrisboden** [boyueluzhipeng/GPT_CodeInterpreter#9]
I have this great project running successfully on replit but am having an issue in my local environment and perhaps you could help.

After installing requirements and setting up the .env file as instructed, Chainlit fires up just fine and I type in hello and get a looping error "Error communicating with OpenAI"
See below

-----------

MacBook-Pro gptcode % chainlit run app.py            
2023-07-30 21:44:04 - Loaded .env file
need_install_package
python_exec
need_rename_file
functions: [{'name': 'need_install_package', 'description': "If the user's question mentions installing packages, and the packages need to be installed,", 'parameters': {'type': 'object', 'properties': {'package_name': {'type': 'string', 'description': 'The name of the package.(required)'}}, 'required': ['package_name']}}, {'name': 'python_exec', 'description': 'A Python shell. Use this to execute python commands in jupyter kernel. Input should be a valid python command.', 'parameters': {'type': 'object', 'properties': {'code': {'type': 'string', 'description': '(str, required):You can write python code here.'}}, 'required': ['code']}}, {'name': 'need_rename_file', 'description': "When the user's question refers to managing files and requires file rename, you can invoke this function.", 'parameters': {'type': 'object', 'properties': {'old_path': {'type': 'string', 'description': 'The old path of the file.(required)'}, 'new_path': {'type': 'string', 'description': 'The new path of the file.(required)'}}, 'required': ['old_path', 'new_path']}}]
2023-07-30 21:44:05 - Your app is available at http://localhost:8000
==================================
hi
==================================
need_install_package
python_exec
need_rename_file
functions: [{'name': 'need_install_package', 'description': "If the user's question mentions installing packages, and the packages need to be installed,", 'parameters': {'type': 'object', 'properties': {'package_name': {'type': 'string', 'description': 'The name of the package.(required)'}}, 'required': ['package_name']}}, {'name': 'python_exec', 'description': 'A Python shell. Use this to execute python commands in jupyter kernel. Input should be a valid python command.', 'parameters': {'type': 'object', 'properties': {'code': {'type': 'string', 'description': '(str, required):You can write python code here.'}}, 'required': ['code']}}, {'name': 'need_rename_file', 'description': "When the user's question refers to managing files and requires file rename, you can invoke this function.", 'parameters': {'type': 'object', 'properties': {'old_path': {'type': 'string', 'description': 'The old path of the file.(required)'}, 'new_path': {'type': 'string', 'description': 'The new path of the file.(required)'}}, 'required': ['old_path', 'new_path']}}]
Error communicating with OpenAI

**Comment by boyueluzhipeng** [comment]
Yes, I often encounter this issue during local testing as well. After extensive testing, I found that the problem originates from the openai library. It is caused by using 'acreate'. When encountering such an issue, you can try running 'chainlit run app_cn.py', which should resolve the problem. However, this is just a workaround solution. Currently, I am also waiting for the official fix to this issue. The impact of using app_cn.py may be that asynchronous implementation in the chat process cannot be achieved effectively.

**Comment by shanmu-raoyunfei** [comment]
同样的问题

**Comment by shanmu-raoyunfei** [comment]
> 是的，我在本地测试时也经常遇到这个问题。经过大量测试，我发现问题源于openai库。这是由于使用“acreate”引起的。遇到此类问题时，您可以尝试运行“chainlit run app_cn.py”，这应该可以解决问题。然而，这只是一个解决方案。目前我也在等待官方修复这个问题。使用app_cn.py的影响可能是聊天过程中无法有效实现异步实现。

一样的问题，还是提示需要安装包
